scripts/pythondistdeps.py

```diff
@@ -111,10 +111,17 @@
             path_item = f
             metadata = FileMetadata(f)
         dist = Distribution.from_location(path_item, dist_name, metadata)
-        # Check if py_version is defined in the file
+        # Check if py_version is defined in the metadata file/directory name
         if not dist.py_version:
-            warn("Version for {!r} has not been found".format(dist), RuntimeWarning)
-            continue
+            # Try to parse the Python version from the path the metadata
+            # resides at (e.g. /usr/lib/pythonX.Y/site-packages/...)
+            import re
+            res = re.search("/python(?P<pyver>\d+\.\d)/", path_item)
```

Comment:
Did you test this RE? If I am not mistaken you need to either put r before the " or quote the backslashes as I did in my previous comment.